core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/RawlsApiService.scala

```diff
@@ -1,121 +1,124 @@
+      case e: Throwable =>
+        complete(StatusCodes.InternalServerError -> ErrorReport(e))
     }
-  )
+  }
 }
 
-trait RootRawlsApiService extends HttpService {
+trait RawlsApiService //(val workspaceServiceConstructor: UserInfo => WorkspaceService, val userServiceConstructor: UserInfo => UserService, val genomicsServiceConstructor: UserInfo => GenomicsService, val statisticsServiceConstructor: UserInfo => StatisticsService, val statusServiceConstructor: () => StatusService, val executionServiceCluster: ExecutionServiceCluster, val appVersion: ApplicationVersion, val googleClientId: String, val submissionTimeout: FiniteDuration, override val workbenchMetricBaseName: String, val samDAO: SamDAO, val swaggerConfig: SwaggerConfig)(implicit val executionContext: ExecutionContext, val materializer: Materializer)
+  extends WorkspaceApiService with EntityApiService with MethodConfigApiService with SubmissionApiService
+  with AdminApiService with UserApiService with BillingApiService with NotificationsApiService
+  with StatusApiService with InstrumentationDirectives with SwaggerRoutes with VersionApiService {
+
+  val workspaceServiceConstructor: UserInfo => WorkspaceService
+  val userServiceConstructor: UserInfo => UserService
+  val genomicsServiceConstructor: UserInfo => GenomicsService
+  val statisticsServiceConstructor: UserInfo => StatisticsService
+  val statusServiceConstructor: () => StatusService
   val executionServiceCluster: ExecutionServiceCluster
   val appVersion: ApplicationVersion
   val googleClientId: String
+  val submissionTimeout: FiniteDuration
+  val workbenchMetricBaseName: String
+  val samDAO: SamDAO
+  val swaggerConfig: SwaggerConfig
 
-  val baseRoute = {
-    path("headers") {
-      get {
-        requestContext => requestContext.complete(requestContext.request.headers.mkString(",\n"))
-      }
-    }
+  implicit val executionContext: ExecutionContext
+  implicit val materializer: Materializer
+
+  def apiRoutes = options { complete(OK) } ~ workspaceRoutes ~ entityRoutes ~ methodConfigRoutes ~ submissionRoutes ~ adminRoutes ~ userRoutes ~ billingRoutes ~ notificationsRoutes
+  def registerRoutes = options { complete(OK) } ~ createUserRoute
+
+  def route: server.Route = (logRequestResult & handleExceptions(RawlsApiService.exceptionHandler)) {
+    swaggerRoutes ~
+    versionRoutes ~
+    statusRoute ~
+    pathPrefix("api") { apiRoutes } ~
+    pathPrefix("register") { registerRoutes }
   }
 
-  private val swaggerUiPath = "META-INF/resources/webjars/swagger-ui/2.2.5"
+  // basis for logRequestResult lifted from http://stackoverflow.com/questions/32475471/how-does-one-log-akka-http-client-requests
+  private def logRequestResult: Directive0 = {
+    def entityAsString(entity: HttpEntity): Future[String] = {
+      entity.dataBytes
+        .map(_.decodeString(entity.contentType.charsetOption.getOrElse(HttpCharsets.`UTF-8`).value))
```

Comment:
in Leo and workbench-libs we use `entity.toStrict(timeout)`. i'm not sure why, i think it might de-stream? @rtitle will know

Reply:
We do both depending on the situation. It feels kind of dirty IMO to call `toStrict` but sometimes you need to.

Reply:
what does strict even mean?

Reply:
It effectively halts the stream as it pulls the entity fully into memory. So if you need to work directly on the entity in a manner which can't be fit into the stream that's what ya gotta do.

Reply:
do you know why is it called strict... the word does not have meaning to me in this context

Reply:
I don’t know why they chose that particular word for the concept, but that’s what it is. It’s a subclass of httpentity

Reply:
I wonder if they named it that way after 'lazy vs. strict'.